From 019a318b2b82bb6cd36f4ecf1eea2a1b599ab528 Mon Sep 17 00:00:00 2001
From: Roman Meshkevich <roman.meshkevich@cogentembedded.com>
Date: Sun, 18 Sep 2016 19:14:00 +0300
Subject: [PATCH] VSPM modify device tree for adapting with VSPM update

From 2d2da562214657572f46abacd11e4a4cf4bf843f Mon Sep 17 00:00:00 2001

This patch file modify device tree of RZ/G1E and RZ/G1M for adapting with VSPM update

Signed-off-by: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
Signed-off-by: Roman Meshkevich <roman.meshekvich>@cogentembedded.com>
---
 arch/arm/boot/dts/r8a7743.dtsi | 44 +++++++++++++-----------------------------
 arch/arm/boot/dts/r8a7745.dtsi | 28 ++++++++-------------------
 2 files changed, 21 insertions(+), 51 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7743.dtsi b/arch/arm/boot/dts/r8a7743.dtsi
index f29673f..7a0782c 100644
--- a/arch/arm/boot/dts/r8a7743.dtsi
+++ b/arch/arm/boot/dts/r8a7743.dtsi
@@ -927,55 +927,37 @@
 		power-domains = <&sysc R8A7743_PD_ALWAYS_ON>;
 		status = "disabled";
 	};
-
-	dmac2d: dma-controller@fea00000 {
-		compatible = "renesas,2ddmac-r8a7743";
-		reg = <0 0xfea00000 0 0x0200>;
+        
+	tddmac@0xfea00000 {
+		compatible = "renesas,vspm-tddmac","renesas,vspm";
+		reg = <0 0xfea00000 0 0x200>;
 		interrupts = <0 285 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&mstp1_clks R8A7743_CLK_2DDMAC>;
 		power-domains = <&sysc R8A7743_PD_ALWAYS_ON>;
 	};
 
-	vsps: vspm@fe928000 {
-		compatible = "renesas,vsps";
-		reg = <0 0xfe928000 0 0x8000>;
+	vsps@fe928000 {
+		compatible = "renesas,vspm-vsps","renesas-vspm";
+		reg = <0 0xfe928000 0 0x7404>;
 		interrupts = <0 267 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&mstp1_clks R8A7743_CLK_VSP1_S>;
 		power-domains = <&sysc R8A7743_PD_ALWAYS_ON>;
-
-		renesas,has-lut;
-		renesas,has-sru;
-		renesas,#rpf = <5>;
-		renesas,#uds = <3>;
-		renesas,#wpf = <4>;
 	};
 
-	vspd0: vspm@fe930000 {
-		compatible = "renesas,vspd0";
-		reg = <0 0xfe930000 0 0x8000>;
+	vspd0@fe930000 {
+		compatible = "renesas,vspm-vspd0","renesas-vspm";
+		reg = <0 0xfe930000 0 0x7404>;
 		interrupts = <0 246 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&mstp1_clks R8A7743_CLK_VSP1_DU0>;
 		power-domains = <&sysc R8A7743_PD_ALWAYS_ON>;
-
-		renesas,has-lif;
-		renesas,has-lut;
-		renesas,#rpf = <4>;
-		renesas,#uds = <1>;
-		renesas,#wpf = <4>;
 	};
 
-	vspd1: vspm@fe938000 {
-		compatible = "renesas,vspd1";
-		reg = <0 0xfe938000 0 0x8000>;
+	vspd1@fe938000 {
+		compatible = "renesas,vspm-vspd1","renesas-vspm";
+		reg = <0 0xfe938000 0 0x7404>;
 		interrupts = <0 247 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&mstp1_clks R8A7743_CLK_VSP1_DU1>;
 		power-domains = <&sysc R8A7743_PD_ALWAYS_ON>;
-
-		renesas,has-lif;
-		renesas,has-lut;
-		renesas,#rpf = <4>;
-		renesas,#uds = <1>;
-		renesas,#wpf = <4>;
 	};
 
 	fdp0: fdpm@fe940000 {
diff --git a/arch/arm/boot/dts/r8a7745.dtsi b/arch/arm/boot/dts/r8a7745.dtsi
index 8f5d11d..58d1157 100644
--- a/arch/arm/boot/dts/r8a7745.dtsi
+++ b/arch/arm/boot/dts/r8a7745.dtsi
@@ -901,40 +901,28 @@
 		};
 	};
 
-	dmac2d: dma-controller@fea00000 {
-		compatible = "renesas,2ddmac-r8a7745";
+	tddmac@fea00000 {
+		compatible = "renesas,vspm-tddmac", "renesas-vspm";
 		reg = <0 0xfea00000 0 0x0200>;
 		interrupts = <0 285 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&mstp1_clks R8A7745_CLK_2DDMAC>;
 		power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 	};
 
-	vsps: vspm@fe928000 {
-		compatible = "renesas,vsps";
-		reg = <0 0xfe928000 0 0x8000>;
+	vsps@fe928000 {
+		compatible = "renesas,vspm-vsps", "renesas-vspm";
+		reg = <0 0xfe928000 0 0x7404>;
 		interrupts = <0 267 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&mstp1_clks R8A7745_CLK_VSP1_S>;
 		power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
-
-		renesas,has-lut;
-		renesas,has-sru;
-		renesas,#rpf = <5>;
-		renesas,#uds = <3>;
-		renesas,#wpf = <4>;
 	};
 
-	vspd0: vspm@fe930000 {
-		compatible = "renesas,vspd0";
-		reg = <0 0xfe930000 0 0x8000>;
+	vspd0@fe930000 {
+		compatible = "renesas,vspm-vspd0", "renesas-vspm";
+		reg = <0 0xfe930000 0 0x7404>;
 		interrupts = <0 246 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&mstp1_clks R8A7745_CLK_VSP1_DU0>;
 		power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
-
-		renesas,has-lif;
-		renesas,has-lut;
-		renesas,#rpf = <4>;
-		renesas,#uds = <1>;
-		renesas,#wpf = <4>;
 	};
 
 	fdp0: fdpm@fe940000 {
-- 
2.7.4

